doc/sphinx/source/backends/executable-api/index.rst

```diff
@@ -4,11 +4,29 @@
 Executable
 ==========
 
+* :ref:`generic_executable`
+* :ref:`intelgpu_executable`
+
+
 The ``compile`` function on an ``Executable`` has more direct methods to 
 actions such as ``validate``, ``call``, ``get_performance_data``, and so on. 
 
 
+.. _generic_executable:
+
+Generic executable
```

Comment:
Same comment as above (on `Generic backend`)

Reply:
It only made sense to title the section when there are more than one, so since we don't need IntelGPU, I'll remove the subsection.